chore(docs): add node tutorial files (#3663)
nrwl/nx: docs/node/tutorial/03-share-code.md

# Step 3: Share Code

> Awesome! The application is working like we expect!

Reviewer: ```suggestion
Awesome! The application is working as expected!
```

> Next, let's work on refactoring a bit. What we're planning to do is move the `Todo` type to it's own library. This would be important if we want to share the type between multiple applications, without having to duplicate the same type everywhere.

Reviewer: ```suggestion
Next, let's work on refactoring a bit. What we're planning to do is move the `Todo` type to it's own library. This is important if we want to share the type between multiple applications, without having to duplicate the same type everywhere.
```

Reviewer: avoid passive voice

**Run the following generator to create a library:**

```bash
nx g @nrwl/workspace:lib data
```

The result should look like this:

```treeview
myorg/
├── apps/
│   └── todos/
├── libs/
│   └── data/
│       ├── src/
│       │   ├── lib/
│       │   │   └── data.ts
│       │   └── index.ts
│       ├── jest.config.js
│       ├── tsconfig.lib.json
│       ├── tsconfig.json
│       └── tsconfig.spec.json
├── tools/
├── nx.json
├── package.json
└── tsconfig.base.json
```

**Copy the type into `libs/data/src/lib/data.ts`.**

```typescript
export type Todo = {
  message: string;
  done: boolean;
};
```

### A note about VS Code :

> If you're using [VS Code](https://code.visualstudio.com/) it may be necessary at this point to restart the TS server so that the new `@myorg/data` package is recognised. This may need to be done **every time a new workspace library is added**.

Reviewer: ```suggestion
If you're using [VS Code](https://code.visualstudio.com/) it may be necessary at this point to restart the TS server so that the new `@myorg/data` package is recognised. This needs to be done **every time a new workspace library is added**.

Use the command pallette with a `*.ts` file open and choose `Restart TS Server`.
```
